### DISCUSSION TITLE
How can I quickly understand the advantages and development direction of symfony6.

### DISCUSSION THREAD

**Discussion by CJayHe** [symfony/symfony#43076] [Q&A] [answered]
How can I quickly understand the advantages and development direction of symfony6.

**Comment by wouterj** [comment, marked as answer]
I don't think there is a need to explain the advantages of a new release (newer code, it'll be maintained for longer, etc.). 

Maybe you mean what is the advantage of removing deprecations in Symfony 6.0? It's good to remove all compatibility layers every so often (Symfony does it once every 2 years). This reduces the work to maintain all these compatibility layers and often also improve performance a bit, as there is no need to check whether a compatibility layer is needed anymore.
It's based, I think, on the "tick-tock" model used by Intel. Push hard for features, then take a break and remove/optimize things, to make room for new features again.

See also https://symfony.com/blog/preparing-your-apps-and-bundles-for-symfony-6 that was published today

"Developer direction" is not a thing in Symfony. There is no roadmap, we're completely contribution driven. If you're passionate about a feature that is missing, you can create the feature and propose it for the upcoming version. The same applies to everything done by core team members: we just contribute features that we are passionate about (or have a need for in our own client projects).
